src/tablestore.rs

```diff
@@ -224,6 +217,23 @@ impl TableStore {
                 _ => SlateDBError::from(e),
             })?;
 
+        if let Some(cache) = self.block_cache.as_ref() {
+            if write_cache {
+                for block in encoded_sst.unconsumed_blocks {
+                    let offset = block.offset;
+                    let block = Arc::new(block.block);
+                    cache
+                        .insert((*id, offset).into(), CachedEntry::with_block(block))
+                        .await;
+                }
+                cache
+                    .insert(
+                        (*id, encoded_sst.info.index_offset).into(),
+                        CachedEntry::with_sst_index(Arc::new(encoded_sst.index)),
+                    )
+                    .await;
+            }
+        }
         self.cache_filter(*id, encoded_sst.info.filter_offset, encoded_sst.filter)
```

Comment:
Interesting that we're already caching the filter. Should this move inside the `if write_cache`? The `cache_filter` method seems somewhat similar...